### DISCUSSION TITLE
`intersect` for groups and modules: embeddings or not...

### DISCUSSION THREAD

**Discussion by fingolfin** [oscar-system/Oscar.jl#1445] [General]
Right now, the `intersect` methods for GAP groups and for `SubQuo` (and presumably for certain other types, too) returns not just the intersection, but also embeddings of the intersection into the objects being intersected. (For GAP groups, these are actually not necessary, but that's not so important right now).

Anyway, I don't like this, as it makes them behave differently compared to other `intersect` methods in Oscar and elsewhere in the Julia ecosystem. Yet of course there are situations where one really needs those embeddings (e.g. I assume this is the case for the `SubQuo` method).

Still, here are some ideas how we could let those `intersect` methods return just the intersection by default:
1. We add an optional `with_embeddings::Bool=false` keyword argument
2. We add `intersect_with_embeddings` variants
3. We finally implement this "canonical maps" system @fieker and me have been talking about for some time: then if `intersect(A,B) = C`, the embeddings would be "registered" as canonical maps `C \to A` resp. `C \to B`, accessed via, say, `canonical_map(C,A)`
4. We just leave it to affected code to implement ad-hoc measures to allow access to embeddings; e.g.  `SubQuo` is attribute storing, so we could simply compute all those embeddings and store them as attributes
5. ... ? other ideas?

The advantage of 1 and 2 is that if the user is not interested in the embeddings, we don't even need to create them, while with 3 and 4 we'd probably have to always create them -- but then again that's not worse than what we do right now.

**Comment by thofma** [comment]
I would be in favour of 2 (and 3), which would align with my preferred solution for https://github.com/oscar-system/Oscar.jl/discussions/1348.
